### PR TITLE
build(jumpserver): 添加 JumpServer 镜像

### PR DESCRIPTION
**Pull request by wojiushixiaobai**

### CODE REVIEW

**Comment by qiangxuhui** [comment]
- cr.loongnix.cn/jumpserver/magnus:v3.10.9

| code | level | alerts |
| --- | --- | --- |
| CIS-DI-0001 | WARN | Last user should not be root |
| CIS-DI-0005 | INFO | export DOCKER_CONTENT_TRUST=1 before docker pull/build |
| CIS-DI-0006 | INFO | not found HEALTHCHECK statement |
| CIS-DI-0008 | INFO | setgid file: grwxr-xr-x usr/sbin/unix_chkpwd |

**Comment by qiangxuhui** [comment]
- cr.loongnix.cn/jumpserver/web:v3.10.9

| code | level | alerts |
| --- | --- | --- |
| CIS-DI-0010 | FATAL | Suspicious ENV key found : NGINX_GPGKEY on |0 /bin/sh -c set -x     && groupadd --system --gid 101 nginx     && useradd --system --gid nginx --no-create-home --home /nonexistent --comment "nginx user" --shell /bin/false --uid 101 nginx     && apt-get update     && apt-get install --no-install-recommends --no-install-suggests -y gnupg1 ca-certificates     &&     NGINX_GPGKEY=573BFD6B3D8FBC641079A6ABABF5BD827BD9BF62;     NGINX_GPGKEY_PATH=/usr/share/keyrings/nginx-archive-keyring.gpg;     export GNUPGHOME="$(mktemp -d)";     found='';     for server in         hkp://keyserver.ubuntu.com:80         pgp.mit.edu     ; do         echo "Fetching GPG key $NGINX_GPGKEY from $server";         gpg1 --keyserver "$server" --keyserver-options timeout=10 --recv-keys "$NGINX_GPGKEY" && found=yes && break;     done;     test -z "$found" && echo >&2 "error: failed to fetch GPG key $NGINX_GPGKEY" && exit 1;     gpg1 --export "$NGINX_GPGKEY" > "$NGINX_GPGKEY_PATH" ;     rm -rf "$GNUPGHOME";     apt-get remove --purge --auto-remove -y gnupg1 && rm -rf /var/lib/apt/lists/*     && dpkgArch="$(dpkg --print-architecture)"     && nginxPackages="         nginx=${NGINX_VERSION}-${PKG_RELEASE}         nginx-module-xslt=${NGINX_VERSION}-${PKG_RELEASE}         nginx-module-geoip=${NGINX_VERSION}-${PKG_RELEASE}         nginx-module-image-filter=${NGINX_VERSION}-${PKG_RELEASE}         nginx-module-njs=${NGINX_VERSION}+${NJS_VERSION}-${PKG_RELEASE}     "     && case "$dpkgArch" in         amd64|arm64)             echo "deb [signed-by=$NGINX_GPGKEY_PATH] https://nginx.org/packages/debian/ buster nginx" > /etc/apt/sources.list.d/nginx.list             && apt-get update             ;;         loongarch64)             echo "deb [signed-by=/usr/share/keyrings/jumpserver-nginx-keyring.gpg] https://dl.cloudsmith.io/public/jumpserver/nginx/deb/debian buster main" > /etc/apt/sources.list.d/nginx.list             && apt-get update             ;;         *)             echo "deb-src [signed-by=$NGINX_GPGKEY_PATH] https://nginx.org/packages/debian/ buster nginx" >> /etc/apt/sources.list.d/nginx.list                         && tempDir="$(mktemp -d)"             && chmod 777 "$tempDir"                         && savedAptMark="$(apt-mark showmanual)"                         && apt-get update             && apt-get build-dep -y $nginxPackages             && (                 cd "$tempDir"                 && DEB_BUILD_OPTIONS="nocheck parallel=$(nproc)"                     apt-get source --compile $nginxPackages             )                         && apt-mark showmanual | xargs apt-mark auto > /dev/null             && { [ -z "$savedAptMark" ] || apt-mark manual $savedAptMark; }                         && ls -lAFh "$tempDir"             && ( cd "$tempDir" && dpkg-scanpackages . > Packages )             && grep '^Package: ' "$tempDir/Packages"             && echo "deb [ trusted=yes ] file://$tempDir ./" > /etc/apt/sources.list.d/temp.list             && apt-get -o Acquire::GzipIndexes=false update             ;;     esac         && apt-get install --no-install-recommends --no-install-suggests -y                         $nginxPackages                         gettext-base                         curl     && apt-get remove --purge --auto-remove -y && rm -rf /var/lib/apt/lists/* /etc/apt/sources.list.d/nginx.list         && if [ -n "$tempDir" ]; then         apt-get purge -y --auto-remove         && rm -rf "$tempDir" /etc/apt/sources.list.d/temp.list;     fi     && ln -sf /dev/stdout /var/log/nginx/access.log     && ln -sf /dev/stderr /var/log/nginx/error.log     && mkdir /docker-entrypoint.d (You can suppress it with --accept-key) |
| CIS-DI-0001 | WARN | Last user should not be root |
| CIS-DI-0005 | INFO | export DOCKER_CONTENT_TRUST=1 before docker pull/build |
| CIS-DI-0006 | INFO | not found HEALTHCHECK statement |
| CIS-DI-0008 | INFO | setuid file: urwxr-xr-x usr/bin/umount |

**Comment by qiangxuhui** [comment]
- cr.loongnix.cn/jumpserver/core:v3.10.9

| code | level | alerts |
| --- | --- | --- |
| CIS-DI-0010 | FATAL | Suspicious filename found : opt/py3/lib/python3.11/site-packages/exchangelib/settings.py (You can suppress it with "-af settings.py") |
| CIS-DI-0001 | WARN | Last user should not be root |
| CIS-DI-0005 | INFO | export DOCKER_CONTENT_TRUST=1 before docker pull/build |
| CIS-DI-0006 | INFO | not found HEALTHCHECK statement |
| CIS-DI-0008 | INFO | setuid file: urwxr-xr-x usr/bin/gpasswd |
| DKL-LI-0003 | INFO | unnecessary file : opt/py3/lib/python3.11/site-packages/ansible_collections/cyberark/conjur/dev/docker-compose.yml  |

**Comment by qiangxuhui** [comment]
- cr.loongnix.cn/jumpserver/lion:v3.10.9

| code | level | alerts |
| --- | --- | --- |
| CIS-DI-0001 | WARN | Last user should not be root |
| CIS-DI-0005 | INFO | export DOCKER_CONTENT_TRUST=1 before docker pull/build |
| CIS-DI-0008 | INFO | setuid file: urwxr-xr-x usr/bin/chfn |

**Comment by qiangxuhui** [comment]
- cr.loongnix.cn/jumpserver/chen:v3.10.9

| code | level | alerts |
| --- | --- | --- |
| CIS-DI-0001 | WARN | Last user should not be root |
| CIS-DI-0005 | INFO | export DOCKER_CONTENT_TRUST=1 before docker pull/build |
| CIS-DI-0006 | INFO | not found HEALTHCHECK statement |
| CIS-DI-0008 | INFO | setgid file: grwxr-xr-x usr/bin/chage |

**Comment by qiangxuhui** [comment]
- cr.loongnix.cn/jumpserver/koko:v3.10.9

| code | level | alerts |
| --- | --- | --- |
| CIS-DI-0001 | WARN | Last user should not be root |
| CIS-DI-0005 | INFO | export DOCKER_CONTENT_TRUST=1 before docker pull/build |
| CIS-DI-0006 | INFO | not found HEALTHCHECK statement |
| CIS-DI-0008 | INFO | setuid file: urwxr-xr-x usr/bin/newgrp |

**Comment by qiangxuhui** [comment]
- cr.loongnix.cn/jumpserver/core:v3.10.9

| code | level | alerts |
| --- | --- | --- |
| CIS-DI-0010 | FATAL | Suspicious filename found : opt/py3/lib/python3.11/site-packages/jedi/settings.py (You can suppress it with "-af settings.py") |
| CIS-DI-0001 | WARN | Last user should not be root |
| CIS-DI-0005 | INFO | export DOCKER_CONTENT_TRUST=1 before docker pull/build |
| CIS-DI-0006 | INFO | not found HEALTHCHECK statement |
| CIS-DI-0008 | INFO | setuid file: urwxr-xr-x usr/bin/passwd |
| DKL-LI-0003 | INFO | unnecessary file : opt/py3/lib/python3.11/site-packages/ansible_collections/community/okd/ci/Dockerfile  |

**Comment by qiangxuhui** [comment]
- cr.loongnix.cn/jumpserver/koko:v3.10.9

| code | level | alerts |
| --- | --- | --- |
| CIS-DI-0001 | WARN | Last user should not be root |
| CIS-DI-0005 | INFO | export DOCKER_CONTENT_TRUST=1 before docker pull/build |
| CIS-DI-0006 | INFO | not found HEALTHCHECK statement |
| CIS-DI-0008 | INFO | setuid file: urwxr-xr-x usr/lib/openssh/ssh-keysign |

**Comment by qiangxuhui** [comment]
- cr.loongnix.cn/jumpserver/chen:v3.10.9

| code | level | alerts |
| --- | --- | --- |
| CIS-DI-0001 | WARN | Last user should not be root |
| CIS-DI-0005 | INFO | export DOCKER_CONTENT_TRUST=1 before docker pull/build |
| CIS-DI-0006 | INFO | not found HEALTHCHECK statement |
| CIS-DI-0008 | INFO | setuid file: urwxr-xr-x usr/bin/chsh |

**Comment by qiangxuhui** [comment]
- cr.loongnix.cn/jumpserver/kael:v3.10.9

| code | level | alerts |
| --- | --- | --- |
| CIS-DI-0001 | WARN | Last user should not be root |
| CIS-DI-0005 | INFO | export DOCKER_CONTENT_TRUST=1 before docker pull/build |
| CIS-DI-0006 | INFO | not found HEALTHCHECK statement |
| CIS-DI-0008 | INFO | setgid file: grwxr-xr-x usr/sbin/unix_chkpwd |

**Comment by qiangxuhui** [comment]
- cr.loongnix.cn/jumpserver/web:v3.10.9

| code | level | alerts |
| --- | --- | --- |
| CIS-DI-0010 | FATAL | Suspicious ENV key found : NGINX_GPGKEY on |0 /bin/sh -c set -x     && groupadd --system --gid 101 nginx     && useradd --system --gid nginx --no-create-home --home /nonexistent --comment "nginx user" --shell /bin/false --uid 101 nginx     && apt-get update     && apt-get install --no-install-recommends --no-install-suggests -y gnupg1 ca-certificates     &&     NGINX_GPGKEY=573BFD6B3D8FBC641079A6ABABF5BD827BD9BF62;     NGINX_GPGKEY_PATH=/usr/share/keyrings/nginx-archive-keyring.gpg;     export GNUPGHOME="$(mktemp -d)";     found='';     for server in         hkp://keyserver.ubuntu.com:80         pgp.mit.edu     ; do         echo "Fetching GPG key $NGINX_GPGKEY from $server";         gpg1 --keyserver "$server" --keyserver-options timeout=10 --recv-keys "$NGINX_GPGKEY" && found=yes && break;     done;     test -z "$found" && echo >&2 "error: failed to fetch GPG key $NGINX_GPGKEY" && exit 1;     gpg1 --export "$NGINX_GPGKEY" > "$NGINX_GPGKEY_PATH" ;     rm -rf "$GNUPGHOME";     apt-get remove --purge --auto-remove -y gnupg1 && rm -rf /var/lib/apt/lists/*     && dpkgArch="$(dpkg --print-architecture)"     && nginxPackages="         nginx=${NGINX_VERSION}-${PKG_RELEASE}         nginx-module-xslt=${NGINX_VERSION}-${PKG_RELEASE}         nginx-module-geoip=${NGINX_VERSION}-${PKG_RELEASE}         nginx-module-image-filter=${NGINX_VERSION}-${PKG_RELEASE}         nginx-module-njs=${NGINX_VERSION}+${NJS_VERSION}-${PKG_RELEASE}     "     && case "$dpkgArch" in         amd64|arm64)             echo "deb [signed-by=$NGINX_GPGKEY_PATH] https://nginx.org/packages/debian/ buster nginx" > /etc/apt/sources.list.d/nginx.list             && apt-get update             ;;         loongarch64)             echo "deb [signed-by=/usr/share/keyrings/jumpserver-nginx-keyring.gpg] https://dl.cloudsmith.io/public/jumpserver/nginx/deb/debian buster main" > /etc/apt/sources.list.d/nginx.list             && apt-get update             ;;         *)             echo "deb-src [signed-by=$NGINX_GPGKEY_PATH] https://nginx.org/packages/debian/ buster nginx" >> /etc/apt/sources.list.d/nginx.list                         && tempDir="$(mktemp -d)"             && chmod 777 "$tempDir"                         && savedAptMark="$(apt-mark showmanual)"                         && apt-get update             && apt-get build-dep -y $nginxPackages             && (                 cd "$tempDir"                 && DEB_BUILD_OPTIONS="nocheck parallel=$(nproc)"                     apt-get source --compile $nginxPackages             )                         && apt-mark showmanual | xargs apt-mark auto > /dev/null             && { [ -z "$savedAptMark" ] || apt-mark manual $savedAptMark; }                         && ls -lAFh "$tempDir"             && ( cd "$tempDir" && dpkg-scanpackages . > Packages )             && grep '^Package: ' "$tempDir/Packages"             && echo "deb [ trusted=yes ] file://$tempDir ./" > /etc/apt/sources.list.d/temp.list             && apt-get -o Acquire::GzipIndexes=false update             ;;     esac         && apt-get install --no-install-recommends --no-install-suggests -y                         $nginxPackages                         gettext-base                         curl     && apt-get remove --purge --auto-remove -y && rm -rf /var/lib/apt/lists/* /etc/apt/sources.list.d/nginx.list         && if [ -n "$tempDir" ]; then         apt-get purge -y --auto-remove         && rm -rf "$tempDir" /etc/apt/sources.list.d/temp.list;     fi     && ln -sf /dev/stdout /var/log/nginx/access.log     && ln -sf /dev/stderr /var/log/nginx/error.log     && mkdir /docker-entrypoint.d (You can suppress it with --accept-key) |
| CIS-DI-0001 | WARN | Last user should not be root |
| CIS-DI-0005 | INFO | export DOCKER_CONTENT_TRUST=1 before docker pull/build |
| CIS-DI-0006 | INFO | not found HEALTHCHECK statement |
| CIS-DI-0008 | INFO | setgid file: grwxr-xr-x usr/sbin/unix_chkpwd |

**Comment by qiangxuhui** [comment]
- cr.loongnix.cn/jumpserver/lion:v3.10.9

| code | level | alerts |
| --- | --- | --- |
| CIS-DI-0001 | WARN | Last user should not be root |
| CIS-DI-0005 | INFO | export DOCKER_CONTENT_TRUST=1 before docker pull/build |
| CIS-DI-0008 | INFO | setuid file: urwxr-xr-x usr/bin/newgrp |

**Comment by qiangxuhui** [comment]
- cr.loongnix.cn/jumpserver/magnus:v3.10.9

| code | level | alerts |
| --- | --- | --- |
| CIS-DI-0001 | WARN | Last user should not be root |
| CIS-DI-0005 | INFO | export DOCKER_CONTENT_TRUST=1 before docker pull/build |
| CIS-DI-0006 | INFO | not found HEALTHCHECK statement |
| CIS-DI-0008 | INFO | setgid file: grwxr-xr-x usr/bin/expiry |

**Comment by qiangxuhui** [comment]
- cr.loongnix.cn/jumpserver/web:v3.10.9

| code | level | alerts |
| --- | --- | --- |
| CIS-DI-0010 | FATAL | Suspicious ENV key found : NGINX_GPGKEY on |0 /bin/sh -c set -x     && groupadd --system --gid 101 nginx     && useradd --system --gid nginx --no-create-home --home /nonexistent --comment "nginx user" --shell /bin/false --uid 101 nginx     && apt-get update     && apt-get install --no-install-recommends --no-install-suggests -y gnupg1 ca-certificates     &&     NGINX_GPGKEY=573BFD6B3D8FBC641079A6ABABF5BD827BD9BF62;     NGINX_GPGKEY_PATH=/usr/share/keyrings/nginx-archive-keyring.gpg;     export GNUPGHOME="$(mktemp -d)";     found='';     for server in         hkp://keyserver.ubuntu.com:80         pgp.mit.edu     ; do         echo "Fetching GPG key $NGINX_GPGKEY from $server";         gpg1 --keyserver "$server" --keyserver-options timeout=10 --recv-keys "$NGINX_GPGKEY" && found=yes && break;     done;     test -z "$found" && echo >&2 "error: failed to fetch GPG key $NGINX_GPGKEY" && exit 1;     gpg1 --export "$NGINX_GPGKEY" > "$NGINX_GPGKEY_PATH" ;     rm -rf "$GNUPGHOME";     apt-get remove --purge --auto-remove -y gnupg1 && rm -rf /var/lib/apt/lists/*     && dpkgArch="$(dpkg --print-architecture)"     && nginxPackages="         nginx=${NGINX_VERSION}-${PKG_RELEASE}         nginx-module-xslt=${NGINX_VERSION}-${PKG_RELEASE}         nginx-module-geoip=${NGINX_VERSION}-${PKG_RELEASE}         nginx-module-image-filter=${NGINX_VERSION}-${PKG_RELEASE}         nginx-module-njs=${NGINX_VERSION}+${NJS_VERSION}-${PKG_RELEASE}     "     && case "$dpkgArch" in         amd64|arm64)             echo "deb [signed-by=$NGINX_GPGKEY_PATH] https://nginx.org/packages/debian/ buster nginx" > /etc/apt/sources.list.d/nginx.list             && apt-get update             ;;         loongarch64)             echo "deb [signed-by=/usr/share/keyrings/jumpserver-nginx-keyring.gpg] https://dl.cloudsmith.io/public/jumpserver/nginx/deb/debian buster main" > /etc/apt/sources.list.d/nginx.list             && apt-get update             ;;         *)             echo "deb-src [signed-by=$NGINX_GPGKEY_PATH] https://nginx.org/packages/debian/ buster nginx" >> /etc/apt/sources.list.d/nginx.list                         && tempDir="$(mktemp -d)"             && chmod 777 "$tempDir"                         && savedAptMark="$(apt-mark showmanual)"                         && apt-get update             && apt-get build-dep -y $nginxPackages             && (                 cd "$tempDir"                 && DEB_BUILD_OPTIONS="nocheck parallel=$(nproc)"                     apt-get source --compile $nginxPackages             )                         && apt-mark showmanual | xargs apt-mark auto > /dev/null             && { [ -z "$savedAptMark" ] || apt-mark manual $savedAptMark; }                         && ls -lAFh "$tempDir"             && ( cd "$tempDir" && dpkg-scanpackages . > Packages )             && grep '^Package: ' "$tempDir/Packages"             && echo "deb [ trusted=yes ] file://$tempDir ./" > /etc/apt/sources.list.d/temp.list             && apt-get -o Acquire::GzipIndexes=false update             ;;     esac         && apt-get install --no-install-recommends --no-install-suggests -y                         $nginxPackages                         gettext-base                         curl     && apt-get remove --purge --auto-remove -y && rm -rf /var/lib/apt/lists/* /etc/apt/sources.list.d/nginx.list         && if [ -n "$tempDir" ]; then         apt-get purge -y --auto-remove         && rm -rf "$tempDir" /etc/apt/sources.list.d/temp.list;     fi     && ln -sf /dev/stdout /var/log/nginx/access.log     && ln -sf /dev/stderr /var/log/nginx/error.log     && mkdir /docker-entrypoint.d (You can suppress it with --accept-key) |
| CIS-DI-0001 | WARN | Last user should not be root |
| CIS-DI-0005 | INFO | export DOCKER_CONTENT_TRUST=1 before docker pull/build |
| CIS-DI-0006 | INFO | not found HEALTHCHECK statement |
| CIS-DI-0008 | INFO | setuid file: urwxr-xr-x usr/bin/newgrp |

**Comment by qiangxuhui** [comment]
- cr.loongnix.cn/jumpserver/lion:v3.10.9

| code | level | alerts |
| --- | --- | --- |
| CIS-DI-0001 | WARN | Last user should not be root |
| CIS-DI-0005 | INFO | export DOCKER_CONTENT_TRUST=1 before docker pull/build |
| CIS-DI-0008 | INFO | setuid file: urwxr-xr-x usr/bin/chfn |

**Comment by qiangxuhui** [comment]
- cr.loongnix.cn/jumpserver/chen:v3.10.9

| code | level | alerts |
| --- | --- | --- |
| CIS-DI-0001 | WARN | Last user should not be root |
| CIS-DI-0005 | INFO | export DOCKER_CONTENT_TRUST=1 before docker pull/build |
| CIS-DI-0006 | INFO | not found HEALTHCHECK statement |
| CIS-DI-0008 | INFO | setuid file: urwxr-xr-x usr/bin/passwd |

File: jumpserver/web/v3.10.9/entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+if [ -f "/etc/init.d/cron" ]; then
+  /etc/init.d/cron start
+fi
+
+until /usr/local/bin/check ${CORE_HOST}/api/health/; do
+    echo "wait for jms_core ${CORE_HOST} ready"
+    sleep 2
+done
+
+nginx -g "daemon off;"
```
Comment (zhangguanzhang) [inline jumpserver/web/v3.10.9/entrypoint.sh:13]: 这个前面不加exec的吗

Comment (wojiushixiaobai) [reply jumpserver/web/v3.10.9/entrypoint.sh:13]: 嗯，不用的

Comment (zhangguanzhang) [reply jumpserver/web/v3.10.9/entrypoint.sh:13]: > 嗯，不用的

不加就不是pid为1，不能接收docker stop信号

Comment (wojiushixiaobai) [reply jumpserver/web/v3.10.9/entrypoint.sh:13]: 基本上都没做这方面的控制，严格的话应该都需要加 exec。
最开始这样做的原因是为了查看进程数的时候方便区分项目，后面就没改回去了。

如果这边要求需要的话我后面再修改一下

**Comment by qiangxuhui** [comment]
- cr.loongnix.cn/jumpserver/chen:v3.10.9

| code | level | alerts |
| --- | --- | --- |
| CIS-DI-0001 | WARN | Last user should not be root |
| CIS-DI-0005 | INFO | export DOCKER_CONTENT_TRUST=1 before docker pull/build |
| CIS-DI-0006 | INFO | not found HEALTHCHECK statement |
| CIS-DI-0008 | INFO | setgid file: grwxr-xr-x usr/bin/expiry |

**Comment by qiangxuhui** [comment]
- cr.loongnix.cn/jumpserver/koko:v3.10.9

| code | level | alerts |
| --- | --- | --- |
| CIS-DI-0001 | WARN | Last user should not be root |
| CIS-DI-0005 | INFO | export DOCKER_CONTENT_TRUST=1 before docker pull/build |
| CIS-DI-0006 | INFO | not found HEALTHCHECK statement |
| CIS-DI-0008 | INFO | setuid file: urwxr-xr-x usr/bin/gpasswd |

**Comment by qiangxuhui** [comment]
- cr.loongnix.cn/jumpserver/magnus:v3.10.9

| code | level | alerts |
| --- | --- | --- |
| CIS-DI-0001 | WARN | Last user should not be root |
| CIS-DI-0005 | INFO | export DOCKER_CONTENT_TRUST=1 before docker pull/build |
| CIS-DI-0006 | INFO | not found HEALTHCHECK statement |
| CIS-DI-0008 | INFO | setgid file: grwxr-xr-x usr/sbin/unix_chkpwd |

**Comment by qiangxuhui** [comment]
- cr.loongnix.cn/jumpserver/lion:v3.10.9

| code | level | alerts |
| --- | --- | --- |
| CIS-DI-0001 | WARN | Last user should not be root |
| CIS-DI-0005 | INFO | export DOCKER_CONTENT_TRUST=1 before docker pull/build |
| CIS-DI-0006 | INFO | not found HEALTHCHECK statement |
| CIS-DI-0008 | INFO | setuid file: urwxr-xr-x usr/bin/umount |

**Comment by qiangxuhui** [comment]
- cr.loongnix.cn/jumpserver/kael:v3.10.9

| code | level | alerts |
| --- | --- | --- |
| CIS-DI-0001 | WARN | Last user should not be root |
| CIS-DI-0005 | INFO | export DOCKER_CONTENT_TRUST=1 before docker pull/build |
| CIS-DI-0006 | INFO | not found HEALTHCHECK statement |
| CIS-DI-0008 | INFO | setuid file: urwxr-xr-x usr/bin/chsh |

**Comment by qiangxuhui** [comment]
- cr.loongnix.cn/jumpserver/web:v3.10.9

| code | level | alerts |
| --- | --- | --- |
| CIS-DI-0010 | FATAL | Suspicious ENV key found : NGINX_GPGKEY on |0 /bin/sh -c set -x     && groupadd --system --gid 101 nginx     && useradd --system --gid nginx --no-create-home --home /nonexistent --comment "nginx user" --shell /bin/false --uid 101 nginx     && apt-get update     && apt-get install --no-install-recommends --no-install-suggests -y gnupg1 ca-certificates     &&     NGINX_GPGKEY=573BFD6B3D8FBC641079A6ABABF5BD827BD9BF62;     NGINX_GPGKEY_PATH=/usr/share/keyrings/nginx-archive-keyring.gpg;     export GNUPGHOME="$(mktemp -d)";     found='';     for server in         hkp://keyserver.ubuntu.com:80         pgp.mit.edu     ; do         echo "Fetching GPG key $NGINX_GPGKEY from $server";         gpg1 --keyserver "$server" --keyserver-options timeout=10 --recv-keys "$NGINX_GPGKEY" && found=yes && break;     done;     test -z "$found" && echo >&2 "error: failed to fetch GPG key $NGINX_GPGKEY" && exit 1;     gpg1 --export "$NGINX_GPGKEY" > "$NGINX_GPGKEY_PATH" ;     rm -rf "$GNUPGHOME";     apt-get remove --purge --auto-remove -y gnupg1 && rm -rf /var/lib/apt/lists/*     && dpkgArch="$(dpkg --print-architecture)"     && nginxPackages="         nginx=${NGINX_VERSION}-${PKG_RELEASE}         nginx-module-xslt=${NGINX_VERSION}-${PKG_RELEASE}         nginx-module-geoip=${NGINX_VERSION}-${PKG_RELEASE}         nginx-module-image-filter=${NGINX_VERSION}-${PKG_RELEASE}         nginx-module-njs=${NGINX_VERSION}+${NJS_VERSION}-${PKG_RELEASE}     "     && case "$dpkgArch" in         amd64|arm64)             echo "deb [signed-by=$NGINX_GPGKEY_PATH] https://nginx.org/packages/debian/ buster nginx" > /etc/apt/sources.list.d/nginx.list             && apt-get update             ;;         loongarch64)             echo "deb [signed-by=/usr/share/keyrings/jumpserver-nginx-keyring.gpg] https://dl.cloudsmith.io/public/jumpserver/nginx/deb/debian buster main" > /etc/apt/sources.list.d/nginx.list             && apt-get update             ;;         *)             echo "deb-src [signed-by=$NGINX_GPGKEY_PATH] https://nginx.org/packages/debian/ buster nginx" >> /etc/apt/sources.list.d/nginx.list                         && tempDir="$(mktemp -d)"             && chmod 777 "$tempDir"                         && savedAptMark="$(apt-mark showmanual)"                         && apt-get update             && apt-get build-dep -y $nginxPackages             && (                 cd "$tempDir"                 && DEB_BUILD_OPTIONS="nocheck parallel=$(nproc)"                     apt-get source --compile $nginxPackages             )                         && apt-mark showmanual | xargs apt-mark auto > /dev/null             && { [ -z "$savedAptMark" ] || apt-mark manual $savedAptMark; }                         && ls -lAFh "$tempDir"             && ( cd "$tempDir" && dpkg-scanpackages . > Packages )             && grep '^Package: ' "$tempDir/Packages"             && echo "deb [ trusted=yes ] file://$tempDir ./" > /etc/apt/sources.list.d/temp.list             && apt-get -o Acquire::GzipIndexes=false update             ;;     esac         && apt-get install --no-install-recommends --no-install-suggests -y                         $nginxPackages                         gettext-base                         curl     && apt-get remove --purge --auto-remove -y && rm -rf /var/lib/apt/lists/* /etc/apt/sources.list.d/nginx.list         && if [ -n "$tempDir" ]; then         apt-get purge -y --auto-remove         && rm -rf "$tempDir" /etc/apt/sources.list.d/temp.list;     fi     && ln -sf /dev/stdout /var/log/nginx/access.log     && ln -sf /dev/stderr /var/log/nginx/error.log     && mkdir /docker-entrypoint.d (You can suppress it with --accept-key) |
| CIS-DI-0001 | WARN | Last user should not be root |
| CIS-DI-0005 | INFO | export DOCKER_CONTENT_TRUST=1 before docker pull/build |
| CIS-DI-0006 | INFO | not found HEALTHCHECK statement |
| CIS-DI-0008 | INFO | setgid file: grwxr-xr-x usr/bin/wall |

**Comment by qiangxuhui** [comment]
- cr.loongnix.cn/jumpserver/chen:v3.10.9

| code | level | alerts |
| --- | --- | --- |
| CIS-DI-0001 | WARN | Last user should not be root |
| CIS-DI-0005 | INFO | export DOCKER_CONTENT_TRUST=1 before docker pull/build |
| CIS-DI-0006 | INFO | not found HEALTHCHECK statement |
| CIS-DI-0008 | INFO | setuid file: urwxr-xr-x usr/bin/newgrp |

**Comment by qiangxuhui** [comment]
- cr.loongnix.cn/jumpserver/core:v3.10.9

| code | level | alerts |
| --- | --- | --- |
| CIS-DI-0010 | FATAL | Suspicious filename found : opt/jumpserver/apps/authentication/backends/saml2/settings.py (You can suppress it with "-af settings.py") |
| CIS-DI-0001 | WARN | Last user should not be root |
| CIS-DI-0005 | INFO | export DOCKER_CONTENT_TRUST=1 before docker pull/build |
| CIS-DI-0006 | INFO | not found HEALTHCHECK statement |
| CIS-DI-0008 | INFO | setuid file: urwxr-xr-x usr/bin/mount |
| DKL-LI-0003 | INFO | unnecessary file : opt/py3/lib/python3.11/site-packages/ansible_collections/community/okd/ci/Dockerfile  |

**Pull request closed without merging**